File: src/main/java/pw/vodes/rimuru/LogRepeater.java
package pw.vodes.rimuru;

import java.util.ArrayList;

import org.javacord.api.entity.message.embed.EmbedBuilder;

public class LogRepeater {
	
	public static ArrayList<EmbedBuilder> embedsToSend = new ArrayList<>();
	
	public static void start() {
		new Thread(() -> {
			while(true) {
				var toSend = new ArrayList<EmbedBuilder>();
				toSend.addAll(embedsToSend);
				embedsToSend.clear();
				for(var embed : toSend) {
					try {
							Main.getConfig().getOtherLogChannel().sendMessage(embed);
					} catch (Exception e) { embedsToSend.add(embed); }
				}
				try {
					Thread.sleep(15000);
				} catch (InterruptedException e) {
					e.printStackTrace();
				}
			}
		}).start();
	}

}

File: src/main/kotlin/pw/vodes/rimurukt/LogRepeater.kt
package pw.vodes.rimurukt

import kotlinx.coroutines.delay
import org.javacord.api.entity.message.embed.EmbedBuilder

object LogRepeater {
    var embedsToSend = mutableListOf<EmbedBuilder>()

    fun start() {
        launchGlobal {
            while (true) {
                val list = embedsToSend.toList()
                embedsToSend.clear()
                list.forEach {
                    try {
                        Main.config.userLogChannel()!!.sendMessage(it)
                    } catch (_: Exception) {
                        embedsToSend.add(it)
                    }
                }
                delay(15000)
            }
        }
    }
}